docs/2.guide/2.directory-structure/1.middleware.md

```diff
@@ -30,7 +30,7 @@ export default defineNuxtRouteMiddleware((to, from) => {
   if (to.params.id === '1') {
     return abortNavigation()
   }
-  if (to.path !== '/') {
+  if (from.path !== '/') {
```

Comment:
would you explain what the issue is here, from your point of view?

Reply:
The previous version with to.path forced the user onto the "/" route no matter where the navigation actually went, the new version makes sure the user isn't already on the "/" route before navigating

Reply:
Don't they both have the same effect?

Reply:
Maybe they do, but the to.path check will make it impossible for users to navigate, as it will force redirect to the "/" route

Reply:
indeed. you would never write this as-is but it's just providing an example of a redirection

Reply:
Even though it's only an example of a redirection, won't it be better to have that example be functional and implementable in real-life applications? It seems a little irrational to provide an unrealistic example. This maintains the 'exampleness' of it while also making it realistic and ready for use in actual websites.

Reply:
Not to mention the docs also had it wrong in the abort-navigation, which I also covered in this PR

Reply:
I'm very up for making this more realistic, but `from` is no more realistic here than `to`, and in most cases will be inaccurate because it isn't guarding the route you are going *to*. We could also just add a comment to explain more.

Reply:
How does it not guard it? The middleware runs with a redirect, so if we see the user is on the same route he's trying to go to, it won't run as it won't redirect. Alternatively, if from is still problematic, we can change it to "if (to.path !== from.path){
    navigateTo(to.path)
}" (if needed can be changed to fullPath)

Reply:
I still don't think that will be right. 

* Current behaviour - correctly redirects any non-`/` routes to `/`: https://stackblitz.com/edit/github-3spxnz
* Proposed behaviour - results in an infinite redirect: https://stackblitz.com/edit/github-3spxnz-qztmiu